src/goMain.ts

```diff
@@ -106,7 +107,10 @@ export function activate(ctx: vscode.ExtensionContext): void {
 		}
 
 		if (vscode.window.activeTextEditor && isGoPathSet()) {
-			runBuilds(vscode.window.activeTextEditor.document, vscode.workspace.getConfiguration('go', vscode.window.activeTextEditor.document.uri));
+			// preload packages so the cache are ready to use
+			loadPackages().then(() => {
```

Comment:
why tie the preload to the initial running of builds? We can run the `getAllPackages()` independent of this correct?

Reply:
Actually doesn't have to tie the initial running build. But it just UI/UX consideration (maybe while I'm not an expert in UX).

`runBuild()` will show "Tests Running".
While `loadPackages()` will show "Load Packages" and disappear when it done.
<img width="433" alt="screen shot 2017-10-04 at 9 11 52 am" src="https://user-images.githubusercontent.com/166053/31157362-8e1c3cb6-a8e4-11e7-9e71-6170f227aec8.png">
1. No separator... and it seems too crowd
2. When "Load Packages" disapear.. it _snaps_ the "Test Running" position to the left.

What do you think?

Reply:
The tests running only shows up if `go.testOnSave` is set to `true`
For now, I dont think we need this messaging at all.

Lets wait for some user feedback.
I also have added the telemetry to get the time taken for the `gopkgs`

We can re-visit this next month by which time I will have the data.